backend/app/vending/__init__.py

```diff
@@ -191,6 +247,176 @@ def get_dashboard_link(login=Depends(login_state)) -> VendingRedirect:
+        currency=vend.currency,
+        appshare=vend.appshare,
+        recommended_donation=vend.recommended_donation,
+        minimum_payment=vend.minimum_payment,
+    )
+
+
+@router.post("app/{appid}/setup")
+def post_app_vending_setup(
+    appid: str, setup: VendingSetup, login=Depends(login_state)
+) -> VendingDescriptor:
+    """
+    Create/update the vending status for a given application.  Returns an error
+    if the appid is not known, or if it's already set up for vending with a
+    user other than the one calling this API.
+
+    If you do not have the right to set the vending status for this application
+    then you will also be refused.
+
+    In addition, if any of the currency or amount values constraints are violated
+    then you will get an error
+    """
+
+    if not login["state"].logged_in():
+        raise VendingError(error="not-logged-in")
+
+    vend = ApplicationVendingConfig.by_appid(db, appid)
+    if not vend:
+        # TODO: Check that the calling user owns the given appid
+        # TODO: Verify that the calling user is onboarded with stripe
+        vend = ApplicationVendingConfig(
+            user=login["user"].id,
+            appid=appid,
+            currency=setup.currency,
+            appshare=setup.appshare,
+            recommended_donation=setup.recommended_donation,
+            minimum_payment=setup.minimum_payment,
+        )
+    else:
+        if vend.user != login["user"].id:
+            raise VendingError(error="user-mismatch")
+        vend.currency = setup.currency
+        vend.appshare = setup.appshare
+        vend.recommended_donation = setup.recommended_donation
+        vend.minimum_payment = setup.minimum_payment
+    try:
+        db.session.add(vend)
+        db.session.commit()
+    except Exception as base_exc:
+        raise VendingError(error="bad-values") from base_exc
+
+    return get_app_vending_status(appid)
+
+
+@router.get("app/{appid}/{currency}/{value}")
+def get_app_vending_split(appid: str, currency: str, value: int) -> VendingSplit:
+    """
+    Retrieve the actual split which would be invoked if the user specified that
+    they wished to spend the given amount on the given appid.  Note this is not
+    the exact shape that an invoice would have since donation vs. purchase may
+    need to be taken into account.
+    """
+    vend = ApplicationVendingConfig.by_appid(db, appid)
+    if not vend:
+        raise VendingError(error="not-found")
+    if vend.currency != currency:
+        raise VendingError(error="bad-currency")
+
+    try:
+        shares = prices.compute_app_shares(value, currency, appid, vend.appshare)
+    except ValueError as val_err:
+        raise VendingError(error="bad-app-share") from val_err
+
+    if shares[0][1] < vend.minimum_payment:
+        raise VendingError(error="bad-value")
+
+    return VendingSplit(status="ok", currency=currency, splits=shares)
+
+
+@router.post("app/{appid}")
+def post_app_vending_status(
```

Comment:
The constraint on this endpoint seems to not be behaving as expected.

Expected behaviour: Payment amount must be greater than minimum payment set for application.

Observed behaviour:

With minimum payment set to 3000, recommended to 3500 a proposed payment with value 3000 is denied (`error: "bad-value"`. Interestingly a value of 3500 is also denied with the same error. A value of 5000 is denied, a value of 6000 is accepted.

Reply:
I tracked down what's going on here, the `get_app_vending_split` method calculates the breakdown and checks if the application's share specifically is greater than or equal to the minimum payment. Whereas elsewhere we (at least I have been) treat the minimum payment as the minimum amount the app developer wants the user to pay (which is then broken down).